Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -7198,7 +7198,8 @@ void WebPage::didCommitLoad(WebFrame* frame)
     
     bool viewportChanged = false;
 
-    m_viewportConfiguration.setCanIgnoreViewportArgumentsToAvoidExcessiveZoom(shouldIgnoreMetaViewport());
+    if (auto* document = coreFrame ? coreFrame->document() : nullptr; document && document->quirks().shouldIgnoreViewportArgumentsToAvoidExcessiveZoom())
+        m_viewportConfiguration.setCanIgnoreViewportArgumentsToAvoidExcessiveZoom(shouldIgnoreMetaViewport());
```

Comment:
Nit - might be good to pull this behind a static helper function to make it slightly less repetitive, e.g.:

```
static void setCanIgnoreViewportArgumentsToAvoidExcessiveZoomIfNeeded(ViewportConfiguration& configuration, Frame* frame)
{
    …
}
```

Reply:
Good idea! Implemented in [81e2290](https://github.com/WebKit/WebKit/pull/15552/commits/81e2290a4e53230b94bc0131d53dd89c8f793c43)